### PR TITLE
Use “Keep A CHANGELOG” format for CHANGELOG.md

### PR DESCRIPTION
**Pull request by Zearin**
It’s not a formal standard, but I think it should be: 

http://keepachangelog.com

It’s basically a collection of sensible things for keeping a changelog meaningful and optimizing human readability.

### CODE REVIEW

**Comment by jkrems** [comment]
Hi! Thanks for the proposal! Interesting project, never heard about it before. I really like that it actually links the versions to the `/compare/` page.

Unfortunately we generate this file semi-automatically using [`npub`](https://github.com/EndangeredMassa/npub). Changing the format without making the change in npub first won't be maintainable. I created https://github.com/EndangeredMassa/npub/issues/29 for discussion.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,58 +1,105 @@
...
+[1.0.7]: https://github.com/groupon/cson-parser/compare/v1.0.7...v1.0.6
+[1.0.6]: https://github.com/groupon/cson-parser/compare/v1.0.6...v1.0.5
+[1.0.5]: https://github.com/groupon/cson-parser/compare/v1.0.5...v1.0.4
+[1.0.4]: https://github.com/groupon/cson-parser/compare/v1.0.4...v1.0.3
+[1.0.3]: https://github.com/groupon/cson-parser/compare/v1.0.3...v1.0.2
+[1.0.2]: https://github.com/groupon/cson-parser/compare/v1.0.2...v1.0.1
+[1.0.1]: https://github.com/groupon/cson-parser/compare/v1.0.1...v1.0.0
+[1.0.0]: https://github.com/groupon/cson-parser/compare/v1.0.0...v0.2.0
+[0.2.0]: https://github.com/groupon/cson-parser/compare/v0.2.0...v0.1.1
+[0.1.1]: https://github.com/groupon/cson-parser/compare/v0.1.1...v0.1.0
```
Comment (johan) [inline CHANGELOG.md]: All of the above links have the tag order backwards, and thus don't work.

Comment (Zearin) [inline CHANGELOG.md]: Aw, crap!  I meant to check those before I submitted this, but I forgot.  Thanks for catching it!

Will fix soon.

**Comment by Zearin** [comment]
> Unfortunately we generate this file semi-automatically using npub. Changing the format without making the change in npub first won't be maintainable. I created EndangeredMassa/npub#29 for discussion.

Cool!  I am unfamiliar with `npub`.  I’ll check it out in the meantime.

File: CHANGELOG.md (hunk 1 above)
Comment (Zearin) [inline CHANGELOG.md]: Fixed!

**Comment by khoomeister** [comment]
If we added PR links for each line item, the version compare link is semi-redundant, plus harder to understand.  Nevertheless it's quite easy to automate so it could be useful for people who don't link to the individual diffs.

**Comment by jkrems** [comment]
@khoomeister Everything that advertises the existence of compare URLs is a net win in my book ;)

**Comment by jkrems** [comment]
Moved to https://github.com/groupon/nlm/issues/10

**Pull request closed without merging**